gprofiler/system_metrics.py

```diff
@@ -0,0 +1,81 @@
+import statistics
+import time
+from collections import deque
+from threading import Event, Thread
+from typing import Deque, List, Optional, Tuple
+
+import psutil
+
+DEFAULT_POLLING_INTERVAL_SECONDS = 5
+
+
+class SystemMetricsMonitor:
+    def __init__(self, max_memory_poll_age_seconds: int, polling_rate_seconds: int = DEFAULT_POLLING_INTERVAL_SECONDS):
+        self._cpu_count = psutil.cpu_count() or 1
+        self._mem_percentages: Deque[float] = deque(maxlen=round(max_memory_poll_age_seconds / polling_rate_seconds))
+        self._last_cpu_poll_time: Optional[float] = None
+        self._last_cpu_utilization_percentages: Optional[Tuple[float, float]] = None
+        self._stop_event = Event()
+        self._thread = Thread(target=self._continuously_polling_memory, args=(polling_rate_seconds,))
+
+        self.get_cpu_utilization()  # Call this once to set the necessary data
+
+    def start(self):
+        self._stop_event.clear()
+        self._thread.start()
```

Comment:
Thread is not reusable after stopping it... so Event doesn't have to be cleared, or thread can be created in here.

Reply:
Made it reusable